runtime/src/bank.rs

```diff
@@ -3579,6 +3594,30 @@ impl Bank {
         self.log_epoch_rewards_sysvar("update");
     }
 
+    /// Create the persistent PDA containing the epoch-rewards data
+    fn create_epoch_rewards_partition_list_account(
+        &self,
+        num_partitions: usize,
+        parent_blockhash: Hash,
+    ) {
+        let epoch_rewards_partition_data = EpochRewardsPartitionDataVersion::V0(PartitionData {
+            num_partitions,
+            parent_blockhash,
+            hasher_kind: HasherKind::Sip13,
+        });
+        let address = get_epoch_rewards_partition_data_address(self.epoch());
+
+        let data_len = bincode::serialized_size(&epoch_rewards_partition_data).unwrap() as usize;
+        let account_balance = self.get_minimum_balance_for_rent_exemption(data_len);
+        let new_account = AccountSharedData::new_data(
+            account_balance,
+            &epoch_rewards_partition_data,
+            &solana_sdk::stake::program::id(),
+        )
+        .unwrap();
+        self.store_account_and_update_capitalization(&address, &new_account);
```

Comment:
how does this work if an account already exists at the given address? like some smart ass sends it some sol in advance

Reply:
Here's the method: https://github.com/solana-labs/solana/blob/22fcffeea84d05111e86cc9e7957cdf9f82f91c2/runtime/src/bank.rs#L6497
If there are more lamports than needed in the existing account, the excess are burned. If less, they are created/supplemented to the new account balance.
It's the same thing other sysvars use.